### DISCUSSION TITLE
Row grouping in data-table

### DISCUSSION THREAD

**Discussion by JuanPedroPontVerges** [shadcn-ui/ui#2809] [Q&A] [answered]
Hello guys.
How can I have row grouping in the current data-table. Meaning that if a click a row, more rows should expand from it? If anyone has an idea please help.

**Comment by lucasfelixs** [comment]
Hi @JuanPedroPontVerges, to be honest, I've never used an expandable line, but to give you a way to solve this, maybe you could use an [accordion component](https://ui.shadcn.com/docs/components/accordion) on the row. I just don't know how you can put other lines inside the accordion, but maybe this is a good way to go.

**Comment by JuanPedroPontVerges** [comment, marked as answer]
So i've figured out a way to have a collapse in a row.


```javascript
<TableBody>
  {table.getRowModel().rows?.length ? (
    table.getRowModel().rows.map((row) => (
      <Collapsible key={row.id} asChild>
        <>
          <CollapsibleTrigger asChild>
            <TableRow data-state={row.getIsSelected() && "selected"} className="!appearance-none">
              {row.getVisibleCells().map((cell) => (
                <TableCell key={cell.id}>
                  {flexRender(cell.column.columnDef.cell, cell.getContext())}
                </TableCell>
              ))}
            </TableRow>
          </CollapsibleTrigger>
          <CollapsibleContent asChild>
            <ExpandedContent dollars={[]} />
          </CollapsibleContent>
        </>
      </Collapsible>
    ))
  ) : (
    <TableRow>
      <TableCell colSpan={columns.length} className="h-24 text-left sm:text-center">
        Sin resultados.
      </TableCell>
    </TableRow>
  )}
</TableBody>
```
          
And this is how the ExpandedContent component looks like:
```javascript
const ExpandedContent: React.FC<ExpandedContentType> = ({ dollars }) => {
  return (
    <>
      <TableRow>
        <TableCell>{1}</TableCell>
        <TableCell>{2}</TableCell>
        <TableCell>Holis!!</TableCell>
      </TableRow>
      <TableRow>
        <TableCell>{1}</TableCell>
        <TableCell>{2}</TableCell>
        <TableCell>Holis!!</TableCell>
      </TableRow>
      <TableRow>
        <TableCell>{1}</TableCell>
        <TableCell>{2}</TableCell>
        <TableCell>Holis!!</TableCell>
      </TableRow>
    </>
  );
};

export default ExpandedContent;
```

And you end up having something like this:

![image](https://github.com/shadcn-ui/ui/assets/31667391/e538a0b8-6a5b-4133-8957-9a22b7607c40)

**Reply by susilthapa** [reply]
Works Great!
One question, how sorting works?

**Reply by JuanPedroPontVerges** [reply]
I dont know how sorting works for default in grouped data structures. You can use the sortingFn in the column definition though, there you can define your own sorting logic

`sortingFn: (rowA, rowB, columnId) => {
      // Your logic inside here
}
`

**Reply by susilthapa** [reply]
@JuanPedroPontVerges Thanks! Will look into that.

**Reply by linaffilomania** [reply]
hi , do you know How to allow only one row to be open at a time

**Comment by linaffilomania** [comment]
hi do you know How to allow only one row to be open at a time?
